libs/ui/inputs/src/lib/dropdown-selector/dropdown-selector.component.ts

```diff
@@ -18,20 +25,76 @@ export type DDChoices<T> = Array<{
   styleUrls: ['./dropdown-selector.component.css'],
   changeDetection: ChangeDetectionStrategy.OnPush,
 })
-export class DropdownSelectorComponent {
+export class DropdownSelectorComponent implements OnInit {
   @Input() title: string
   @Input() showTitle = true
   @Input() ariaName: string
   @Input() choices: DDChoices<unknown>
   @Input() selected: any
+  @Input() maxRows: number
   @Input() extraClass = ''
   @Output() selectValue = new EventEmitter<any>()
+  @ViewChild('overlayOrigin') overlayOrigin: CdkOverlayOrigin
+  @ViewChild(CdkConnectedOverlay) overlay: CdkConnectedOverlay
+  overlayOpen = false
+  overlayWidth = 'auto'
+  overlayMaxHeight = 'none'
+  overlayPositions: ConnectedPosition[] = [
+    {
+      originX: 'start',
+      originY: 'bottom',
+      overlayX: 'start',
+      overlayY: 'top',
+      offsetY: 8,
+    },
+    {
+      originX: 'start',
+      originY: 'top',
+      overlayX: 'start',
+      overlayY: 'bottom',
+      offsetY: -8,
+    },
+  ]
+  get selectedChoice(): Choice {
+    return (
+      this.choices.find((choice) => choice.value === this.selected) ??
+      this.choices[0]
+    )
+  }
 
   get id() {
     return this.title.toLowerCase().replace(/[^a-z]+/g, '-')
   }
 
+  getChoiceLabel(): string {
+    return this.selectedChoice.label
+  }
+
+  ngOnInit(): void {
+    if (!this.maxRows) this.maxRows = 6
```

Comment:
Maybe set a const with value 6 at top of component, to get the hardcoded value outside the code.